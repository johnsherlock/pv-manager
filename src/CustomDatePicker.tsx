import moment from 'moment';
import { forwardRef } from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

interface DatePickerDivProps {
  value: string;
  onClick: () => void;
}

const DatePickerDiv = forwardRef<HTMLDivElement, DatePickerDivProps>(({ value, onClick }, ref) => (
  <div className="custom-date-picker" onClick={onClick} ref={ref}>
    {value}
  </div>
));

interface CustomDatePickerProps {
  selectedDate: moment.Moment;
  onChange: (moment: moment.Moment) => any;
}

const CustomDatePicker = ({ selectedDate, onChange }: CustomDatePickerProps) => {

  return (
    <DatePicker
      selected={selectedDate.toDate()}
      onChange={(date) => { console.log(date); onChange(moment(date)); }}
      placeholderText="Select a date"
      dateFormat="EEE do MMM yyyy"
      minDate={new Date(2023, 0, 20)}
      maxDate={new Date()}
      customInput={<DatePickerDiv value={selectedDate.toString()} onClick={() => {}}/>}
    />
  );
};

export default CustomDatePicker;