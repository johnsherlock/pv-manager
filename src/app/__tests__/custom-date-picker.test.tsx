import { render } from '@testing-library/react';
import moment from 'moment';
import React from 'react';
import CustomDatePicker from '../custom-date-picker';

describe('CustomDatePicker', () => {
  test('renders correctly', () => {
    const selectedDate = moment('2023-02-19');
    const onChange = jest.fn();
    const { container } = render(<CustomDatePicker selectedDate={selectedDate} onChange={onChange} dispatch={() => {}} scale={'day'} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});