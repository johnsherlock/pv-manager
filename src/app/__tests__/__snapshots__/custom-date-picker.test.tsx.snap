// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDatePicker renders correctly 1`] = `
<div
  class="react-datepicker-wrapper"
>
  <div
    class="react-datepicker__input-container"
  >
    <span
      aria-live="polite"
      class="react-datepicker__aria-live"
      role="alert"
    />
    <div
      class="custom-date-picker"
    >
      Sun 19th Feb 2023
    </div>
  </div>
</div>
`;
